### PR TITLE
Reduce API surface area

### PR DESCRIPTION
**Pull request by colin-higgins**
resolves https://github.com/Particular/NServiceBus.RavenDB/issues/

- GatewayMessage (This would suck to obsolete)
- RavenDbSagaStorage (No need to obsolete)

Please review @Particular/transport-maintainers

### CODE REVIEW

File: src/NServiceBus.RavenDB/SagaPersister/RavenDbSagaStorage.cs [hunk 1]
```diff
@@ -5,7 +5,7 @@
     /// <summary>
     ///     RavenDB Saga Storage.
     /// </summary>
-    public class RavenDbSagaStorage : Feature
+    class RavenDbSagaStorage : Feature
```
Comment (SimonCropp) [inline src/NServiceBus.RavenDB/SagaPersister/RavenDbSagaStorage.cs]: generally we dont have xml doco for internal classes.

**Comment by DavidBoike** [comment]
I think we can internalize GatewayMessage too. 

And then, is it best practice to leave obsoleted stubs for the classes we remove from the public API? And how would you do that for GatewayMessage since it's a document stored by Raven?

**Comment by SimonCropp** [comment]
> is it best practice to leave obsoleted stubs for the classes we remove from the public API

Depends on the chance of someone using said public api. I know some stuff in the core was public, but because of how things fit together there was no real way for people to use them. so they were just made internal.

But this decision can only really be made by people intimate with use cases.

Having said that it is low effort to leave obsolete stubs around for a major

**Comment by SimonCropp** [comment]
> how would you do that for GatewayMessage since it's a document stored by Raven

does raven need a public class? is there any use case for people to query that class?

**Comment by DavidBoike** [comment]
Raven does not need public and there's no reason for customer to query on it. However to leave an obsoleted stub behind you'd need to move the real class somewhere else, and THAT Raven would object to.

**Comment by SimonCropp** [comment]
@DavidBoike ahh i understand. yep in that case just make it internal

**Comment by colin-higgins** [comment]
Should we leave a comment GatewayMessage explaining?

**Comment by DavidBoike** [comment]
Currently TC is failing build with:

> Fody: Cannot process 'NServiceBus.PersistenceExtentions`1<NServiceBus.RavenDBPersistence> NServiceBus.RavenDbSettingsExtensions::UseSharedSession(NServiceBus.PersistenceExtentions`1<NServiceBus.RavenDBPersistence>,System.Func`1<Raven.Client.IDocumentSession>)'. The assembly version 6.0.0 is equal to or greater than version specified in 'RemoveInVersion' 5.0.0. The member should be removed or 'RemoveInVersion' increased.

So it's trying to build with AssemblyVersion of 6, when it should really be 4. I noticed in the branch it still had NextVersion.txt rather than GitVersionConfig.yaml. So I tried rebasing locally and building, but then I get the same error in Visual Studio, so I don't want to push that.

So in my local, GitVersionTask is 3.4.1 (latest stable), and GitVersionConfig.yaml looks like this:

```
assembly-versioning-scheme: Major
next-version: 4.0
branches:
  develop:
    tag: alpha
```

What's wrong here? @SimonCropp can you help?

**Comment by SimonCropp** [comment]
@JakeGinnivan can u look at this one?

**Comment by JakeGinnivan** [comment]
The issue is
```
INFO [04/14/16 19:58:35:06] Base version used: Merge message 'Merge pull request #127 from Particular/upgrade-v6.0.0-unstable1293
```

We really need https://github.com/GitTools/GitVersion/issues/538 to be completed so we can exclude parts of history which are causing issues

**Comment by DavidBoike** [comment]
Great, a commit from 4 months ago. And there's no way we can change the commit message without requiring a massive force push right?

**Comment by SimonCropp** [comment]
@JakeGinnivan is is possible to make the regex for that commit message parsing a little stricter. we would never want a message of that format to bump the version

**Comment by SimonCropp** [comment]
> And there's no way we can change the commit message without requiring a massive force push right?

correct

**Comment by SimonCropp** [comment]
although it is not a "massive force push" any more than any other force push

**Comment by DavidBoike** [comment]
@bording and I determined that a version string of "-v6.0" was enough to improperly bump the base version to the incorrect value, so we decided that one solution would be to create a way to have GitVersion completely ignore merge commit messages as a strategy for determining the base version.

I created a patch for GitVersion that does just that. Via the YAML config, you can add this bit, and then the strategy class responsible for getting those merge commit messages gets skipped:

```
disabled-base-version-strategies:
  - MergeMessageBaseVersionStrategy
```

It is in no way suitable yet for a PR (1 test, 0 docs) and I'm not 100% sure that it's the correct strategy long term. So, I pushed it to [my fork](https://github.com/DavidBoike/GItVersion/tree/disable-base-version-strategies, built it myself, and published it as a prerelease package to our MyGet feed. 

I updated this PR to use the new version from MyGet, and by inspecting the build log, it's getting the correct version. However, new problem in TeamCity: 

> C:\BuildAgent\work\c4d62ce02b983878\src\packages\NuGetPackager.0.5.5\build\NuGetPackager.targets(6, 5): error MSB4044: The "CreatePackages" task was not given a value for the required parameter "Version".

I'm not sure if this is a different problem or a side-effect of my work.

@SimonCropp @andreasohlund @JakeGinnivan Any suggestions on how to proceed?

**Comment by danielmarbach** [comment]
You need nuget packager 0.6.0. only this one uses the proper msbuild property

**Comment by danielmarbach** [comment]
Can you explain why the existing commit-message-incrementing: Disabled did not work?

**Comment by DavidBoike** [comment]
@danielmarbach Now that you mention it I only remember it failing on TC when I tried it. I should circle back and ensure it wasn't OK but failing on not having nuget packager 0.6.0.

**Comment by danielmarbach** [comment]
Local release build with msbuild highest log level helps

**Comment by JakeGinnivan** [comment]
@danielmarbach `commit-message-incrementing` only stops the +semver: major style messages. Not merge messages.

@SimonCropp yeah, we can change the regex. But I think the ignore commits option would be better

**Comment by danielmarbach** [comment]
Was more an example. Not a solution :)

File: src/NServiceBus.RavenDB/SagaPersister/RavenDbSagaStorage.cs [hunk 2]
```diff
@@ -2,22 +2,13 @@
 {
     using NServiceBus.SagaPersisters.RavenDB;
 
-    /// <summary>
-    ///     RavenDB Saga Storage.
-    /// </summary>
-    public class RavenDbSagaStorage : Feature
+    class RavenDbSagaStorage : Feature
```
Comment (DavidBoike) [inline src/NServiceBus.RavenDB/SagaPersister/RavenDbSagaStorage.cs]: Let's obsolete this one here, and move it to a different namespace.

Wonder if we should move all the features to NServiceBus.RavenDB or something for consistency anyway? This seems like a very odd namespace to use, not being inside NServiceBus.RavenDB.

**Comment by DavidBoike** [comment]
With a different build (so that GitVersionTask experimental and NuGetPackager version requirements don't fight) we are now building successfully.

**Pull request merged**